dbms/src/Storages/Page/V3/Universal/UniversalPageStorage.cpp

```diff
@@ -510,6 +502,21 @@ PS::V3::CPDataWriteStats UniversalPageStorage::dumpIncrementalCheckpoint(const U
     }
 
     last_checkpoint_sequence = snap->sequence;
+
+    GET_METRIC(tiflash_storage_checkpoint_seconds, type_dump_checkpoint_snapshot).Observe(dump_snapshot_seconds);
+    GET_METRIC(tiflash_storage_checkpoint_seconds, type_dump_checkpoint_data).Observe(dump_data_seconds);
+    GET_METRIC(tiflash_storage_checkpoint_seconds, type_upload_checkpoint).Observe(upload_seconds);
+    LOG_DEBUG(log,
+              "Checkpoint result: files={}, dump_snapshot={:.2f}s, dump_data={:.2f}s, upload={:.2f}s, "
+              "total={:.2f}s, upload_sequence={}, incremental_data_bytes={}, rewrite_data_bytes={}",
```

Comment:
```suggestion
              "Checkpoint result: files={}, dump_snapshot={:.3f}s, dump_data={:.3f}s, upload={:.3f}s, "
              "total={:.3f}s, upload_sequence={}, incremental_data_bytes={}, rewrite_data_bytes={}",
```
`.3f` could be better because it correspond to `ms`